packages/maskbook/src/plugins/FileService/database.ts

```diff
@@ -8,13 +8,23 @@ type TaggedTypes = FileInfo
 const Database = createPluginDatabase<TaggedTypes>(identifier)
 
 export async function getRecentFiles() {
-    const files: FileInfo[] = await asyncIteratorToArray(Database.iterate('arweave'))
-    files.sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime())
-    return files.slice(0, 4)
+    const skynet: FileInfo[] = await asyncIteratorToArray(Database.iterate('skynet'))
+    const arweave: FileInfo[] = await asyncIteratorToArray(Database.iterate('arweave'))
+    if (skynet != null) {
+        const files = skynet
+        files.sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime())
+        return files.slice(0, 4)
+    } else if (arweave != null) {
+        const files = arweave
+        files.sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime())
+        return files.slice(0, 4)
+    }
 }
 
 export async function getFileInfo(checksum: string) {
-    return Database.get('arweave', checksum)
```

Comment:
You need first check `arweave`, if include this checksum then return it
otherwise failover to `skynet`, etc